server/src/player/init.rs

```diff
@@ -43,21 +46,38 @@ impl<'a> System<'a> for PlayerInitSystem {
             mut entity_types,
             mut metadata,
             level,
+            config,
         ) = data;
 
         // Run through events
         for event in events.read(&mut self.join_event_reader.as_mut().unwrap()) {
+            // Load player data
```

Comment:
Rather than using `Option` in the player data struct and then using `Result::unwrap_or_default()`, I would recommend matching on the result from `load_player_data` like this (although this code won't compile):

```rust
let (position, gamemode) = match feather_core::player_data::load_player_data(uuid) {
    Ok(data) => (data.pos, data.gamemode),
    Err(e) => (position!(level.x, level.y, level.z), Gamemode::from_string(default_gamemode)),
};
```